sympy/core/power.py

```diff
@@ -212,11 +212,11 @@ def __new__(cls, b, e, evaluate=None):
                 return S.One
             else:
                 # recognize base as E
-                if not e.is_Atom and b is not S.Exp1 and b.func is not exp_polar:
+                if not e.is_Atom and b is not S.Expl and not(instance(b,exp_polar)):
```

Comment:
It is `Exp1`(the number one, not alphabet l) and did you mean `not isinstance(b, exp_polar)`?